compiler/src/iree/compiler/InputConversion/MHLO/test/transformation_pipeline.mlir

```diff
@@ -44,7 +44,7 @@ func.func @mhloElementwiseOps(%arg0 : tensor<4xf32>) -> tensor<4xf32> {
 
 func.func @interleavedDot(%arg0 : tensor<4x4xf32>) -> tensor<4x4xf32> {
   %0 = mhlo.add %arg0, %arg0 : tensor<4x4xf32>
-  %1 = "mhlo.dot"(%0, %arg0) : (tensor<4x4xf32>, tensor<4x4xf32>) -> tensor<4x4xf32>
+  %1 = "stablehlo.dot"(%0, %arg0) : (tensor<4x4xf32>, tensor<4x4xf32>) -> tensor<4x4xf32>
   %2 = mhlo.multiply %1, %arg0 : tensor<4x4xf32>
   return %2 : tensor<4x4xf32>
 }
```

Comment:
Seems like we should have a test case dedicated to a program that is entirely `stablehlo`? This `mhlo.dot` -> `stablehlo.dot` change just looks incidental...

Reply:
Sure, doesn't make much difference & interleaving should work. Given these will be mostly StableHLO as the pure interchange format in future, I'll just flip this to be entirely stablehlo.

Reply:
Yeah, I'm thinking about this from the angle of "does IREE support stable HLO? point me at the test that proves it does". If we eventually move more tests / input programs to stablehlo instead of mhlo then that'd be great. For now, a single unit test here at least shows that the code works.